### ISSUE TITLE
Multi-wifi automated testing

### ISSUE THREAD

**Issue by alsegno** [liquidinvestigations/liquidinvestigations#102]
Per our discussion yesterday, we would like to simplify the overall configuration of our system by always having two wifi cards.

It looks relatively simple to do a PoC of these configurations by using emulated cards via:

https://wireless.wiki.kernel.org/en/users/drivers/mac80211_hwsim

This will allow us to write automated tests of our wireless functionality, too, which is a huge win.

Alex, do you see any drawbacks to doing a PoC and automated testing using the mac80211_hwsim module?

**Comment by mgax** [comment]
`mac80211_hwsim` - that's so cool. But I'm concerned it will take a lot of effort to automate these tests.

**Comment by alsegno** [comment]
The basic tests are 'working' now but need some more documentation.

We have a python function now that will create a hostapd.conf file based on ssid, password, interface. This hostapd.conf contains a minimal set of options that we may want to expand.

We have a minimal wpa_supplicant.conf that just specifies ssid, password.

We have one test that sets up the emulated radios, starts the AP based on the hostapd.conf, then connects via wpa_supplicant.

It's bundled with a Vagrantfile that sets up a machine and installs required dependencies.

Once the API wifi (re)configuration endpoints are implemented, we'll be able to easily integrate into some further tests.

Next tasks for this:

 - Nicely formatted README.md files everywhere
 - Complete verification of the 'simple case'

**Comment by mgax** [comment]
That's pretty sweet! Did you push the code anywhere? It would be nice to have this emulation stack functioning when we implement the hotspot/client configuration for a node. And, if we manage to write automated tests, we can run them in our shiny new Jenkins server.

**Comment by alsegno** [comment]
Yeah, it's here:

https://github.com/liquidinvestigations/wifi-test

I need to spend some time writing an overall test strategy, cleaning up the README, documenting the python code for hostapd.conf creation, some stuff like that, so it's very much a WIP.

**Comment by gabriel-v** [comment]
blocked on API implementation #114

**Comment by gabriel-v** [comment]
also blocked on #105

**Comment by gabriel-v** [comment]
blocked, moving to backlog
